From 9794f2b103d25138f18e95c17d2cb4bb66638269 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Tue, 18 Dec 2018 16:35:00 +0800
Subject: [PATCH] gpu: mail-400: instead of __GFP_REPEAT with
 __GFP_RETRY_MAYFAIL

According to upstream commit dcda9b04713c("mm, tree wide: replace
__GFP_REPEAT by __GFP_RETRY_MAYFAIL with more useful semantic").
Change __GFP_REPEAT into __GFP_RETRY_MAYFAIL.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../driver/src/devicedrv/mali/linux/mali_memory_os_alloc.c              | 2 +-
 .../driver/src/devicedrv/mali/linux/mali_osk_notification.c             | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/linux/mali_memory_os_alloc.c b/linux/mali_memory_os_alloc.c
index 08fa2fb..3379f4f 100755
--- a/linux/mali_memory_os_alloc.c
+++ b/linux/mali_memory_os_alloc.c
@@ -202,7 +202,7 @@ int mali_mem_os_alloc_pages(mali_mem_os_mem *os_mem, u32 size)
 	/* Allocate new pages, if needed. */
 	for (i = 0; i < remaining; i++) {
 		dma_addr_t dma_addr;
-		gfp_t flags = __GFP_ZERO | __GFP_REPEAT | __GFP_NOWARN | __GFP_COLD;
+		gfp_t flags = __GFP_ZERO | __GFP_RETRY_MAYFAIL | __GFP_NOWARN| __GFP_COLD;
 		int err;
 
 #if defined(CONFIG_ARM) && !defined(CONFIG_ARM_LPAE)
diff --git a/linux/mali_osk_notification.c b/linux/mali_osk_notification.c
index 5867841..be6a0a4 100755
--- a/linux/mali_osk_notification.c
+++ b/linux/mali_osk_notification.c
@@ -57,7 +57,7 @@ _mali_osk_notification_t *_mali_osk_notification_create(u32 type, u32 size)
 	_mali_osk_notification_wrapper_t *notification;
 
 	notification = (_mali_osk_notification_wrapper_t *)kmalloc(sizeof(_mali_osk_notification_wrapper_t) + size,
-			GFP_KERNEL | __GFP_HIGH | __GFP_REPEAT);
+			GFP_KERNEL | __GFP_HIGH | __GFP_RETRY_MAYFAIL);
 	if (NULL == notification) {
 		MALI_DEBUG_PRINT(1, ("Failed to create a notification object\n"));
 		return NULL;
-- 
2.7.4

